dwc_quick_check.py

```diff
@@ -0,0 +1,269 @@
+    # Or do we need to disambiguate Errors from Warnings in the HTML report?
+    if "scientificName" not in df.columns:
+        logging.info("Missing scientificName.")
+        return
+
+    logging.info("🐠 Verifying taxonomy with WoRMS API (this may take a moment)...")
+    # TODO: Should this be exact or fuzzy?
+    # For example, we have 'polychaeta spp. 1' and 'polychaeta spp. 2' there.
+    unique_names = df["scientificName"].dropna().unique().tolist()
+    # WoRMS AphiaRecordsByMatchNames endpoint
+    url = "https://www.marinespecies.org/rest/AphiaRecordsByMatchNames"
+
+    # Process in chunks of 50 to avoid timeouts
+    chunk_size = 50
+    # invalid_taxa = []  # TODO: Unused
+    unmatched_taxa = []
+
+    # TODO: Gemini thinks this is matlab!?
+    for i in range(0, len(unique_names), chunk_size):
+        chunk = unique_names[i : i + chunk_size]
+        try:
+            # Construct query parameters
+            params = {"scientificnames[]": chunk, "marine_only": "true"}
+            response = requests.get(url, params=params)
+
+            if response.status_code == 200:
+                results = response.json()
+
+                # The API returns a list of lists (one list per name queried)
+                for original_name, matches in zip(chunk, results):
+                    if not matches:
+                        unmatched_taxa.append(original_name)
+                    else:
+                        # Check if accepted
+                        match = matches[0]  # Take best match
+                        if match["status"] != "accepted":
+                            logging.info(
+                                f'WARNING", f"Taxon "{original_name}" is {match["status"]}. Accepted name: {match["valid_name"]}, {match["url"]}'
+                            )
+            else:
+                logging.info("WARNING", f"WoRMS API Error: {response.status_code}")
+
+        except Exception as e:
+            logging.info(f"Error connecting to WoRMS: {e}")
+
+        # Polite delay
+        time.sleep(0.5)
+
+    if unmatched_taxa:
+        logging.info(
+            "WARNING", f"Taxa not found in WoRMS: {', '.join(unmatched_taxa[:10])}..."
+        )
+
+
+for name, df, cols in zip(
+    ["event", "occurrence", "emof"],
+    [df_event, df_occurrence, df_emof],
+    [required_event_columns, required_occurrence_columns, required_emof_columns],
+):
+    # TODO: When using namedtuple we can skip tests that won't pass, like checking for depth in dfs that should not have it.
+    logging.info(f"Started {name}.")
+    res = [
+        check_required_columns(df, columns=cols),
+        check_null_values(df, columns=cols),
+        check_coordinates(df),
+        check_depth_consistency(df),
+    ]
+    logging.info(res)
+    logging.info(f"Finished {name}.")
+
+    check_scientific_names(df)
+
+check_merge_tables(df_event=df_event, df_occurrence=df_occurrence, df_emof=df_emof)
```

Comment:
This is the main driver for now. The difference is that I'm trying to run as many tests as possible before the merge so we can return a more meaningful report to the user. We will need to skip a few tests for certain tables. I'll work on that later.